tokio/src/runtime/mod.rs

```diff
@@ -172,6 +172,128 @@
+//! When the collection is empty, the thread will go to sleep until a task is
+//! added to the collection.
+//!
+//! However, the above is not sufficient to guarantee a well-behaved runtime.
+//! For example, the runtime might have a single task that is always ready to be
+//! polled, and poll that task every time. This is a problem because it starves
+//! other tasks by not polling them. To solve this, Tokio provides the following
+//! fairness guarantee:
+//!
+//! > Under the following two assumptions:
+//! >
+//! > * There is some number `MAX_TASKS` such that the total number of tasks on
+//! >   the runtime never exceeds `MAX_TASKS`.
+//! > * There is some number `MAX_POLL` such that calling [`poll`] on any task
+//! >   spawned on the runtime returns within `MAX_POLL` time units.
+//! >
+//! > Then, it is guaranteed that tasks are scheduled fairly, that is, there is
+//! > some number `MAX_DELAY` such that when a task becomes ready, it will be
+//! > polled by the runtime within `MAX_DELAY` time units.
+//!
+//! Other than the above fairness guarantee, there is no guarantee about the
+//! order in which tasks are polled. There is also no guarantee that the runtime
+//! is equally fair to all tasks. For example, if the runtime has two tasks A
+//! and B that are both ready, then the runtime may poll A five times before it
+//! polls B. This is the case even if A yields using [`yield_now`]. All that is
+//! guaranteed is that it will poll B eventually.
+//!
+//! Normally, tasks are polled only if they have notified the runtime by waking
+//! their waker. However, this is not guaranteed, and Tokio may poll tasks that
+//! have not notified the runtime under some circumstances. This is called a
+//! spurious wakeup.
+//!
+//! ## IO and timers
+//!
+//! Beyond just polling tasks, the runtime must also manage IO resources and
+//! timers. It does this by periodically checking whether there are any IO
+//! resources or timers that are ready, and notifying the relevant task so that
+//! it will be polled.
+//!
+//! These checks are performed periodically between polling tasks. Under the
+//! same assumptions as the previous fairness guarantee, Tokio guarantees that
+//! it will notify tasks with an IO or timer event within some maximum number of
+//! time units.
+//!
+//! ## Current behavior (current thread runtime)
+//!
+//! This section describes how the [current thread runtime] behaves today. This
+//! behavior may change in future versions of Tokio.
+//!
+//! The current thread runtime maintains two FIFO queues of tasks that are ready
+//! to be polled: the global queue and the local queue. The runtime will prefer
+//! to choose the next task to poll from the local queue, and will only pick a
+//! task from the global queue if the local queue is empty, or if it has picked
+//! a task from the local queue 31 times in a row. The number 31 can be
+//! changed using the [`global_queue_interval`] setting.
```

Comment:
The only reason this is still the case is nobody has applied the MT behavior here 😅